test/object-store/c_api/c_api.cpp

```diff
@@ -1056,6 +1056,83 @@ TEST_CASE("C API", "[c_api]") {
         CHECK(realm_changed_callback_called);
     }
 
+    SECTION("realm refresh registering callback outside transaction") {
+        bool realm_refresh_callback_called = false;
+        auto token = cptr(realm_add_realm_refresh_callback(
+            realm,
+            [](void* userdata) {
+                *reinterpret_cast<bool*>(userdata) = true;
+            },
+            &realm_refresh_callback_called, [](void*) {}));
+        realm_begin_write(realm);
+        realm_commit(realm);
+        CHECK_FALSE(realm_refresh_callback_called);
+    }
+
+    SECTION("realm refresh registering callback in transaction") {
```

Comment:
can you add some tests around registering during a read transaction and a frozen Realm?